File: src/main/java/com/giant/auth/doc/SignInSuccessResponseDoc.java
package com.giant.auth.doc;

import com.giant.auth.dto.response.SignInResponseDto;
import com.giant.common.api.doc.success.SuccessWithResultResponseDoc;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;

@Schema(name = "SignInSuccessResponse")
@Getter
public class SignInSuccessResponseDoc
        extends SuccessWithResultResponseDoc<SignInResponseDto> {

    @Schema(implementation = SignInResponseDto.class)
    private final SignInResponseDto result = null;
}

File: src/main/kotlin/com/giant/module/auth/doc/SignInSuccessResponseDoc.kt
package com.giant.module.auth.doc

import com.giant.common.api.doc.success.SuccessWithResultResponseDoc
import com.giant.module.auth.dto.response.SignInResponseDto
import io.swagger.v3.oas.annotations.media.Schema

@Schema(name = "SignInSuccessResponse")
class SignInSuccessResponseDoc: SuccessWithResultResponseDoc<SignInResponseDto>() {
    @get:Schema(implementation = SignInResponseDto::class)
    override var result: SignInResponseDto? = null
}